index.js

```diff
@@ -11,11 +11,21 @@ function toObject(val) {
 	return Object(val);
 }
 
-function isFunc(val) {
-	return typeof val === 'function';
+function assignKey(to, from, key) {
+	var val = from[key];
+
+	if (typeof val === 'undefined' || val === null) {
+		return;
+	}
+
+	if (typeof to[key] === 'undefined' || !isObj(val)) {
```

Comment:
Yes, I see and that's a good point.

On Thu, Oct 8, 2015 at 10:14 AM, Michael Mayer <notifications@github.com>
wrote:

> In index.js
> <https://github.com/purposeindustries/deep-assign/pull/1#discussion_r41486584>
> :
>
> > @@ -11,11 +11,21 @@ function toObject(val) {
> >  	return Object(val);
> >  }
> >
> > -function isFunc(val) {
> > -	return typeof val === 'function';
> > +function assignKey(to, from, key) {
> > +	var val = from[key];
> > +
> > +	if (typeof val === 'undefined' || val === null) {
> > +		return;
> > +	}
> > +
> > +	if (typeof to[key] === 'undefined' || !isObj(val)) {
>
> This only concerns properties explicitly set:
>
> deepAssign({foo: undefined}, {foo: …});
>                  ^^^^^^^^^    ^^^
>
> deepAssign({foo: null}, {foo: …});
>                  ^^^^    ^^^
>
> But not the following cases:
>
> deepAssign({}, {foo: …}); // no Error!
> deepAssign({foo: undefined, bar: null}, {xxx: …}); // no Error!
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/purposeindustries/deep-assign/pull/1/files#r41486584>.
>